### PR TITLE
Memento creation

### PR DESCRIPTION
**Pull request by EreliaCorp**

### CODE REVIEW

File: includes/design_pattern/spk_memento.hpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+	private:
+		std::vector<Snapshot> _states;
+		int _currentState = -1;
+
+		void _load(Snapshot& p_snapshot, TType& p_state)
+		{
+			p_snapshot >> p_state;
+			p_snapshot.reset();
+		}
```
Comment (EreliaCorp) [inline includes/design_pattern/spk_memento.hpp:22]: We do not want the DataBuffer to store the object at all. We want the Snapshot to contain defined section of the object, via a secondary classe Memento::Object or something. This object must describe what to do when saving or loading snapshot

File: includes/design_pattern/spk_memento.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		 * @param p_state Value to save.
+		 * @note If there is a state after the current one, it will be erased.
+		 */
+		void save(const TType& p_state)
+		{
+			if (_currentState != _states.size() - 1)
+				_states.erase(_states.begin() + _currentState + 1, _states.end());
+
+			Snapshot newSnapshot;
+			newSnapshot << p_state;
```
Comment (EreliaCorp) [inline includes/design_pattern/spk_memento.hpp:37]: We need to use previously describe Memento::Object

Comment (EreliaCorp) [reply includes/design_pattern/spk_memento.hpp:37]: We probably should have a _save method

File: playground/src/main.cpp [hunk 3]
```diff
@@ -1,14 +1,118 @@
...
+	{
+		p_buffer << p_object._toRemember << p_object._toRememberToo;
+		return p_buffer;
+	}
+
+	friend spk::DataBuffer& operator>>(spk::DataBuffer& p_buffer, Object& p_object)
+	{
+		p_buffer >> p_object._toRemember >> p_object._toRememberToo;
+		return p_buffer;
+	}
```
Comment (EreliaCorp) [inline playground/src/main.cpp:35]: Those two methods should be the _onSave and _onLoad method of the Memento::Object

**Review by EreliaCorp — COMMENTED** [review]
Need to create the Memento::Object

**Review by Neyune — CHANGES_REQUESTED** [review]
Apply Erelia recommandation

**Review by EreliaCorp — COMMENTED** [review]
Is that the problem

**Pull request closed without merging**